test/module/irohad/torii/processor/transaction_processor_test.cpp

```diff
@@ -15,72 +15,403 @@
+ * @when transactions compose proposal which is sent to peer
+ * communication service @and all transactions composed the block
+ * @then for every transaction STATEFUL_VALID status is returned
+ */
+TEST_F(TransactionProcessorTest, TransactionProcessorBlockCreatedTest) {
+  TxSetType txs;
+  for (size_t i = 0; i < proposal_size; i++) {
+    auto tx = TestTransactionBuilder().txCounter(i).build();
+    txs.insert(tx);
+    status_map[tx.hash()] = iroha::model::TransactionResponse::NOT_RECEIVED;
+  }
+
+  auto wrapper = make_test_subscriber<CallExact>(
+      tp->transactionNotifier(), proposal_size * 2);  // every transaction is
+                                                      // notified that it is
+                                                      // stateless valid and
+                                                      // then stateful valid
+  wrapper.subscribe([this](auto response) {
+    auto resp = static_cast<TransactionResponse &>(*response);
+    auto hash = shared_model::crypto::Hash(resp.tx_hash);
+    status_map[hash] = resp.current_status;
+  });
+
+  blockTest(txs);
+  ASSERT_TRUE(wrapper.validate());
+
+  validateStatuses(
+      txs, iroha::model::TransactionResponse::STATEFUL_VALIDATION_SUCCESS);
+}
+
+/**
+ * @given transaction processor
+ * @when transactions compose proposal which is sent to peer
+ * communication service @and all transactions composed the block @and were
+ * committed
+ * @then for every transaction COMMIT status is returned
+ */
+TEST_F(TransactionProcessorTest, TransactionProcessorOnCommitTest) {
+  TxSetType txs;
+  for (size_t i = 0; i < proposal_size; i++) {
+    auto tx = TestTransactionBuilder().txCounter(i).build();
+    txs.insert(tx);
+    status_map[tx.hash()] = iroha::model::TransactionResponse::NOT_RECEIVED;
+  }
+
+  auto wrapper = make_test_subscriber<CallExact>(
+      tp->transactionNotifier(),
+      proposal_size * 3);  // evey transaction is notified that it is first
+                           // stateless valid, then stateful valid and
+                           // eventually committed
+  wrapper.subscribe([this](auto response) {
+    auto resp = static_cast<TransactionResponse &>(*response);
+    auto hash = shared_model::crypto::Hash(resp.tx_hash);
+    status_map[hash] = resp.current_status;
+  });
+
+  commitTest(txs, txs);
+  ASSERT_TRUE(wrapper.validate());
+
+  validateStatuses(txs, iroha::model::TransactionResponse::COMMITTED);
+}
+
+/**
+ * @given transaction processor
+ * @when transactions compose proposal which is sent to peer
+ * communication service @and some transactions became part of block, while some
+ * were not committed
+ * @then for every transaction from block COMMIT status is returned @and for
+ * every transaction not from block STATEFUL_INVALID_STATUS was returned
+ */
+TEST_F(TransactionProcessorTest, TransactionProcessorInvalidTxsTest) {
+  TxSetType proposal_txs;
+  for (size_t i = 0; i < proposal_size; i++) {
```

Comment:
Making a custom loops which will add specific transaction to lists will simplify the tests, so that `TxSetType` will not be required.